src/recwizard/modules/kbrd/tokenizer_kbrd_gen.py

```diff
@@ -1,19 +1,62 @@
+import os
+import json
 from typing import List
 
 from recwizard.tokenizer_utils import BaseTokenizer
 from recwizard.utility.utils import WrapSingleInput
 
+from .tokenizer_nltk import KBRDWordTokenizer
+from .tokenizer_kbrd_rec import KBRDRecTokenizer
+
+
 class KBRDGenTokenizer(BaseTokenizer):
+    init_word_list_file = "nltk_word_list.json"
 
-    @WrapSingleInput
-    def decode(
-            self,
-            ids,
+    def __init__(
+        self,
+        word_tokenizer: BaseTokenizer = None,
+        entity_tokenizer: BaseTokenizer = None,
+        **kwargs,
+    ):
+        self.word_tokenizer = word_tokenizer
+        self.entity_tokenizer = entity_tokenizer
+        tokenizers = [self.word_tokenizer, self.entity_tokenizer]
+
+        if word_tokenizer is None or entity_tokenizer is None:
+            raise ValueError("word_tokenizer and entity_tokenizer should not be None.")
+
+        super().__init__(tokenizers=tokenizers, **kwargs)
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *args, **kwargs):
```

Comment:
could use get_init_kwargs to save the init word list to the hf hub.

Reply:
Thank you, that would be much cleaner. I managed to get it working today and will push it once I finish the docstrings.

Reply:
Hey @Kevin99z, I use `get_init_kwargs` now and finish the docstring part for our code. Please review it when you have time, thanks!

Reply:
LGTM